src/rational/q/from.rs

```diff
@@ -249,3 +343,105 @@ mod tests_from_str {
+        let q_2 = Q::try_from((&(numerator as u16), &(denominator as i16))).unwrap();
+        let q_3 = Q::try_from((&(numerator as u32), &(denominator as i32))).unwrap();
+        let q_4 = Q::try_from((&(numerator as u64), &(denominator as i64))).unwrap();
+        let q_5 = Q::try_from((&Z::from(numerator), &Z::from(denominator))).unwrap();
+
+        assert_eq!(q_1, q_2);
+        assert_eq!(q_1, q_3);
+        assert_eq!(q_1, q_4);
+        assert_eq!(q_1, q_5);
+    }
+
+    /// Ensure that creating [`Q`] works with small integer parameters.
+    #[test]
+    fn working_small() {
+        let numerator = 10;
+        let denominator = 15;
+        let numerator_z = Z::from(numerator);
+        let denominator_z = Z::from(denominator);
+
+        let q_1 = Q::try_from((&numerator, &denominator)).unwrap();
+        let q_2 = Q::try_from((&(numerator * 2), &(denominator * 2))).unwrap();
+        let q_3 = Q::try_from((&numerator_z, &denominator_z)).unwrap();
+
+        assert_eq!(q_1, q_2);
+        assert_eq!(q_1, q_3);
+    }
+
+    /// Ensure that large parameters work (FLINT uses pointer representation).
+    #[test]
+    fn working_large() {
+        let numerator = u64::MAX;
+        let denominator = u64::MAX - 1;
+        let numerator_z = Z::from(numerator);
+        let denominator_z = Z::from(denominator);
+
+        let q_1 = Q::try_from((&numerator, &denominator)).unwrap();
+        let q_2 = Q::try_from((&numerator_z, &denominator_z)).unwrap();
+
+        assert_eq!(q_1, q_2);
+    }
+
+    /// Test with zero denominator (not valid -> should lead to an error)
+    #[test]
+    fn divide_by_zero() {
+        let numerator = 10;
+        let denominator = 0;
+
+        let new_q = Q::try_from((&numerator, &denominator));
+
+        assert!(new_q.is_err());
+    }
+
+    /// Test with either negative denominator or numerator
+    #[test]
+    fn negative_small() {
+        let numerator = 10;
+        let denominator = -1;
+
+        let q_1 = Q::try_from((&numerator, &denominator)).unwrap();
+        let q_2 = Q::try_from((&-numerator, &-denominator)).unwrap();
+
+        assert_eq!(q_1, q_2);
+    }
+
+    /// Test with negative numerator and denominator. Should be the same as both positive
+    #[test]
+    fn numerator_and_denominator_negative_small() {
+        let numerator = 10;
+        let denominator = -1;
+
+        let q_1 = Q::try_from((&numerator, &denominator)).unwrap();
+        let q_2 = Q::try_from((&-numerator, &-denominator)).unwrap();
+
+        assert_eq!(q_1, q_2);
+    }
+
+    /// Test with negative numerator and denominator. Should be the same as both positive
+    #[test]
+    fn numerator_and_denominator_negative_large() {
+        let numerator = i64::MAX;
+        let denominator = i64::MAX - 1;
+
+        let q_1 = Q::try_from((&numerator, &denominator)).unwrap();
+        let q_2 = Q::try_from((&-numerator, &-denominator)).unwrap();
+
+        assert_eq!(q_1, q_2);
+    }
```

Comment:
These tests check more than you describe. Please adapt the comment (only needed in the first case) and function name accordingly